mutex.go

```diff
@@ -91,15 +73,31 @@ func (m *mutex) Lock() {
 		if err == nil {
 			res.Body.Close()
 			if res.StatusCode == 200 {
-				return
+				// Check for timeout before returning.
+				okToReturn = true
```

Comment:
Also realized, I didn't have a goroutine any more, so the attempt to lock was not running parallel with the select waiting for timeout. I think I've fixed that now, but let me know if you spot anything questionable. The tests now pass and it gives successful return precedence over timeout. PTAL